nvme-topology.c

```diff
@@ -426,10 +426,10 @@ static int legacy_list(struct nvme_topology *t)
 			continue;
 		ret = 0;
 
-		fd = open(path, O_RDONLY);
+		fd = open_global_device(path);
 		if (fd > 0) {
 			nvme_identify_ctrl(fd, &c->id);
-			close(fd);
+			close_global_device();
```

Comment:
open_global_device() always return 0 for pax devices. The close under this condition only closes global device of rc type.

Reply:
Updated with `>=`